### PR TITLE
migration to Azle 0.20.1 with the Express.js approach

### PR DESCRIPTION
**Pull request by taras-hk**

### CODE REVIEW

**Review by Emmanuel-umeh — APPROVED** [review]
LGTM

**Review by Jonath-z — APPROVED** [review]
LGTM

**Comment by lastmjs** [comment]
Azle 0.20.2 has been released with a simplified installation experience, I suggest looking at The Azle Book to understand what to change. Mac users may have issues, we're still in the first few days of this change with a couple known problems from users so far.

**Comment by lastmjs** [comment]
There's also an AUTO_RELOAD environment variable that developers will probably love, you might want to document that

File: README.md
Comment (lastmjs) [inline README.md:1]: Please look at the latest changes in The Azle Book and the hello_world of express examples. Azle 0.20.2 introduces a simplified installation experience

File: README.md [hunk 1]
```diff
@@ -65,23 +82,29 @@ where `message_board` is the name of the canister.
...
   "description": "Internet Computer message board application",
   "dependencies": {
-    "@dfinity/agent": "^0.19.2",
-    "@dfinity/candid": "^0.19.2",
-    "azle": "^0.18.6",
-    "uuid": "^9.0.0"
+    "@dfinity/agent": "^0.21.4",
+    "@dfinity/candid": "^0.21.4",
+    "azle": "^0.20.1",
+    "body-parser": "^1.20.2",
```
Comment (lastmjs) [inline README.md]: You shouldn't need to install body-parser, express.json() or express.text() just works

File: dfx.json
Comment (lastmjs) [inline dfx.json:1]: Look at the latest dfx.json files in the hellow_world or other relevant examples, you'll probably want to set candid_gen to http for a quicker compilation time and fewer possibilities for errors during deployment, also take a look at the other new fields

File: src/index.ts [hunk 2]
```diff
@@ -1,28 +1,7 @@
...
-    attachmentURL: text
-});
-
-const Error = Variant({
-    NotFound: text,
-    InvalidPayload: text,
-});
+import { Server, StableBTreeMap, ic } from 'azle';
+import express, { Request, Response } from 'express';
+import bodyParser from 'body-parser';
```
Comment (lastmjs) [inline src/index.ts]: Shouldn't need body-parser

File: tsconfig.json
Comment (lastmjs) [inline tsconfig.json:1]: I'm not sure the strict property initialization is needed, don't remember, might want to look into it

File: src/index.ts [hunk 3]
```diff
@@ -31,62 +10,79 @@ const Error = Variant({
...
+class Message {
+    id: string;
+    title: string;
+    body: string;
+    attachmentURL: string;
+    createdAt: Date;
+    updatedAt: Date | null
+}
+
+const messagesStorage = StableBTreeMap<string, Message>(0);
```
Comment (lastmjs) [inline src/index.ts:32]: I would recommend declaring this variable with let to indicate mutability, just a subjective best practice I have adopted

File: src/index.ts [hunk 4]
```diff
@@ -31,62 +10,79 @@ const Error = Variant({
...
+
+export default Server(() => {
+    const app = express();
+    app.use(bodyParser.json());
+    
+    app.post("/messages", (req: Request, res: Response) => {
```
Comment (lastmjs) [inline src/index.ts]: I don't believe it's necessary to put the static types explicitly on each req and res parameter, they are inferred.

**Review by lastmjs — CHANGES_REQUESTED** [review]

**Pull request merged**